### ISSUE TITLE
Current build problems ... 907:d0c3e3b2fc12

### ISSUE THREAD

**Issue by lsces** [mkoppanen/imagick#72]
My reference build is still the 3.1.2 version from my PHP5.4.20 base.

I'm in the process of trying to build a compatible 5.6.x base to cross check, but currently it files to compile with 
imagick_class.c:3249:30: error: 'IS_TRUE' undeclared - quick hack for PHP5 and it compiles, but while the resulting package loads, phpinfo() crashes. Spoke too soon - my PHP5 builds will not run.

A couple of tests failed on the PHP7 build, and runs, but http://php7.lsces.org.uk/phpinfo.php is still not showing version or list of extensions.

**Comment by Danack** [comment]
Can you say what system you're compiling on?

**Comment by Danack** [comment]
Additionally, can you run the tests and give the output from test 026_phpinf which will be in the file tests//documents/projects/github/mkoppanen-imagick/tests/026_phpinfo.out which was added to check that phpinfo is running correctly.

**Comment by lsces** [comment]
Can't get imagick back up again on http://lsces.org.uk/phpinfo.php even cloning a working copy from a production site such as http://lsces.crg.uk/phpinfo.php - currently using 'imagick module version' 3.1.2 which you will see above is displayed as @PACKAGE_VERSION@ and yes the formats list is empty but should have 218 entries.

On the PHP7 compile I get
PASS Test Imagick module hasn't broken phpinfo [tests/026_phpinfo.phpt]
FAIL Test Imagick, rotationalBlurImage [tests/113_Imagick_rotationalBlurImage_basic.phpt] 
FAIL Test Imagick, setOption [tests/130_Imagick_setOption_case1.phpt] (with a dump)
FAIL Test Tutorial, deconstructGif [tests/237_Tutorial_deconstructGif_basic.phpt]

I'm working on top of SUSE ... 12.3 still on production, and 13.1 in development. There was a page with all the config details, but not sure it's up to date. Next step is to do a clean build using ... next step is to run 'php-src_PHP-5.6.5' in parallel with my PHP7 build but it's conserning that the cross reference development build has been messed up :(

**Comment by Danack** [comment]
I have fixed the supported formats not being displayed. And Travis is now running all the tests

> FAIL Test Imagick, rotationalBlurImage [tests/113_Imagick_rotationalBlurImage_basic.phpt] 

This is probably due to having an old version of ImageMagick that doesn't include this function. I have changed the test to be skipped if it isn't present.

> FAIL Test Imagick, setOption tests/130_Imagick_setOption_case1.phpt
This appears to be a bug in libjpeg where it fails to save an image as a jpeg correctly. It seems to happen on Travis as well, but I can't reproduce locally so have added an XFAIL for now, until I can investigate

> FAIL Test Tutorial, deconstructGif [tests/237_Tutorial_deconstructGif_basic.phpt]

This fails on Travis as well with the test timing out. Which is odd as it takes less than a second on my machine. I am investigating further but have added an XFAIL for this test for now.

As the actual issue with the formats supported listing is fixed, I'm going to close the issue. Feel free to reopen this if you see it not fixed for you, otherwise please open new issues in a separate issue so that I can track them separately.

fyi the way I'm developing is just to have the source for both PHP5.6 and "7" in separate directories, and then to switch between them I just need to 'make install' in the PHP directory, and then in the Imagick directory run `phpize`, `./configure`, `make clean` and `make install` to switch between versions.

It's not perfect, but it seem to be reasonably reliable way of working.

**Comment by lsces** [comment]
What version was rotationalBlurImage added? suse13.1 has 6.8.6-9 2014-11-13 but that date does not match the imagemagick date for 6.8.6-9 ... 13.2 has 6.8.9-8 but I don't need the new functions.
PHP7 fails on that test while PHP5.6.5 skips it on my setup.
Both display list of types on phpinfo, but @PACKAGE_VERSION@  still populates the version?

**Comment by Danack** [comment]
It was done in this change http://trac.imagemagick.org/changeset/15117 which would putit in version 6.8.8-9 I believe.

> but @PACKAGE_VERSION@ still populates the version? 

Yes, apparently PACKAGE_VERSION only gets populated with an actual version when it is built through the Pecl package manager.

**Comment by lsces** [comment]
I'm getting a similar problem between the code from git and the distributed copies with many files not matching due to a populated ID :(
I'm due to upgrade to SUSE13.2 but not sure if now is the right time ... 13.1 is nice and stable now.
